app/web/features/badges/hooks.ts

```diff
@@ -0,0 +1,41 @@
+import { badgesKey, badgeUsersKey } from "features/queryKeys";
+import { RpcError } from "grpc-web";
+import { ListBadgeUsersRes } from "proto/api_pb";
+import { Badge } from "proto/resources_pb";
+import { useInfiniteQuery, useQuery } from "react-query";
+import { service } from "service";
+
+export const useBadges = () => {
+  const { data, ...rest } = useQuery(badgesKey, () =>
+    service.resources.getBadges().then((result) =>
+      result.badgesList.reduce(
+        (badgesResult, badge) => {
+          badgesResult.badges[badge.id] = badge;
+          return badgesResult;
```

Comment:
Could you switch this over to async await syntax instead? The `.then` is a bit outdated IMO. 

It means you'll need to turn the first into its own variable but that also is more readable.